Add static documentation site for gStore

hrz6976 opened the pull request:
<img width="908" alt="a106b96a44a888e033a1fc5846357e6" src="https://github.com/pkumod/gStore/assets/46741383/8a51ed03-a3ed-4ade-bfa7-38e9ca6ab8a9">

### Note
This PR is still under development and not ready for review yet. Some changes requires further testing and tuning, and have not been pushed to GitHub.

**The snapshot of latest changes is available on cloudflare pages:** https://gstore-docs.pages.dev

### Changes in this PR
* Add static documentation site for the gStore database.
* Requires basically zero changes to the markdown files in `docs`.
* Requires zero lines of code for future updates, table of contents can be edited in `_sidebar.md` in intuitive markdown syntax.

### TODOs
* Restructure the current user manual (sad that a markdown Chinese version is not available)
* Fix UI glitches (the current site generator only respects standard markdown syntax)
* Add CI automatic deployments

### DISCLAIMER
* This PR is a part of the course project in PKU's graph data management course.

hrz6976 committed:
docs: add static documentation site

hrz6976 committed:
docs: update chinese docs

hrz6976 committed:
docs: add license compatibilities with 3rd

hrz6976 committed:
docs: dynamic page title

hrz6976 committed:
docs: fix ui glitches

hrz6976 commented:
Also checked license compatibilities with the following libraries with [LicenseRec](https://licenserec.com/):
- [Boost](https://www.boost.org/) 
- [zlib](https://www.zlib.net/) 
- [OpenSSL](https://www.openssl.org/)
- [indicators](https://github.com/p-ranav/indicators) 
- [rapidjson](http://rapidjson.org/)
- [log4cplus](https://log4cplus.github.io/log4cplus/) 
- [curl](https://curl.se/docs/copyright.html)
- [gomp](https://gcc.gnu.org/projects/gomp/)
- [jemalloc](https://jemalloc.net/) 
- [Antlr4](https://www.antlr.org/) 
- [workflow](https://github.com/sogou/workflow)

hrz6976 committed:
docs: add badges to README

hrz6976 committed:
docs: fix README rendering issues

hrz6976 committed:
docs: update zh-cn README

lsvih committed:
Remove out-of-date docs.

lsvih committed:
Add latest user manual as embedded pdf.

lsvih committed:
Add links to official website.

lsvih commented:
WIP，文档需与官方的 [中文文档](https://www.gstore.cn/pcsite/index.html#/documentation) 和 [英文文档](https://en.gstore.cn/#/enDocs) 同步，需要一点时间

hrz6976 committed:
docs: simplify README

make readme more like a readme from a mature open-source database, e.g. github.com/milvus-io/milvus

hrz6976 committed:
docs: add 404 page

zxlchina commented:
您的邮件我已收到，谢谢~~~
